### ISSUE TITLE
sql: `column_default` in `information_schema.columns` is incompatible with postgres

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#87774]
**Describe the problem**

The content of `column_default` is using a format that confuses postgres-compatible apps.
(Found [here](https://github.com/piccolo-orm/piccolo/issues/607))

There are 2 different problems:

1. its syntax uses type annotations where none are expected.
2. the type annotation is also incompatible with the actual type of the column.

**To Reproduce**

In PostgreSQL:

```
> create table mytable(x smallint default 0);
> select data_type, column_default  from information_schema.columns where table_name = 'mytable' and column_name = 'x';
 data_type | column_default
-----------+----------------
 smallint  | 0
```

In CockroachDB:

```
> create table mytable(x smallint default 0);
> select data_type, column_default  from information_schema.columns where table_name = 'mytable' and column_name = 'x';
  data_type | column_default
------------+-----------------
  smallint  | 0:::INT8
```

cc @rafiss for triage.

Jira issue: CRDB-19526

Epic CRDB-17785

**Comment by rafiss** [comment]
https://github.com/cockroachdb/cockroach/pull/94153 resolves this

**Comment by knz** [comment]
fixed by #94153 indeed
